src/realm/sync/noinst/client_history_impl.cpp

```diff
@@ -78,15 +78,23 @@ void ClientHistory::set_client_reset_adjustments(
 
         util::compression::CompressMemoryArena arena;
         util::AppendBuffer<char> compressed;
-        for (auto& [changeset, version] : recovered_changesets) {
-            uploadable_bytes += changeset.size();
-            auto i = size_t(version - m_sync_history_base_version);
-            util::compression::allocate_and_compress_nonportable(arena, changeset, compressed);
-            m_arrays->changesets.set(i, BinaryData{compressed.data(), compressed.size()}); // Throws
+        // Server version is updated for *every* entry in the sync history to ensure that server versions don't
```

Comment:
I think doing this as two separate loops is a bit clearer (and it's trivially faster, but that doesn't matter):

```
        for (auto& [changeset, version] : recovered_changesets) {
            uploadable_bytes += changeset.size();
            auto i = size_t(version - m_sync_history_base_version);
            util::compression::allocate_and_compress_nonportable(arena, changeset, compressed);
            m_arrays->changesets.set(i, BinaryData{compressed.data(), compressed.size()}); // Throws
        }
        for (size_t i = 0, j = 0, size = m_arrays->remote_versions.size(); i < size; ++i) {
            m_arrays->remote_versions.set(i, server_version.version);
        }
```

Reply:
I initially implemented it like that, then changed my mind. I will make the change.